solutions/instances/main.tf

```diff
@@ -288,8 +363,8 @@ resource "time_sleep" "wait_for_authorization_policy" {
 }
 
 # Data source to account settings for retrieving cross account id
-data "ibm_iam_account_settings" "iam_account_settings" {
-  count = local.apply_auth_policy
+data "ibm_iam_account_settings" "iam_cos_account_settings" {
+  provider = ibm.cos
```

Comment:
this might be a breaking change. I would keep the count and the resource name what it was but just add the provider config in it